library/Garden/EventManager.php

```diff
@@ -287,27 +285,26 @@ private function callMetaHandlers($event, array $args, $result) {
      *
      * @param string $event The name of the event to fire.
      * @param mixed $value The value to pass into the filter.
+     * @param array $args Any arguments the event takes.
      * @return mixed The result of the chained event or `$value` if there were no handlers.
      */
-    public function fireFilter($event, $value) {
+    public function fireFilter($event, $value, ...$args) {
         $handlers = $this->getHandlers($event);
         if (empty($handlers) && empty($this->handlers[self::EVENT_META])) {
             return $value; // gotcha, return value
         }
 
-        $callArgs = $args = array_slice(func_get_args(), 1);
-
+        $result = $value;
         foreach ($handlers as $callback) {
-            $value = call_user_func_array($callback, $callArgs);
-            $callArgs[0] = $value;
+            $result = call_user_func($callback, $result, ...$args);
```

Comment:
TIL that you can unpack variadic arguments!
http://php.net/manual/en/functions.arguments.php#example-145
@beckyvb @initvector @linc